airflow/executors/kubernetes_executor.py

```diff
@@ -545,6 +545,26 @@ def _labels_to_key(self, labels):
             return None
 
         with create_session() as session:
+            # check if we can find the task directly before scanning
+            # every task with the execution_date
+            task = (
+                session
+                .query(TaskInstance)
+                .filter_by(task_id=task_id, dag_id=dag_id, execution_date=ex_time)
+                .first()
+            )
+            if task:
+                self.log.info(
+                    'Found matching task %s-%s (%s) with current state of %s',
+                    task.dag_id, task.task_id, task.execution_date, task.state
+                )
+                return (dag_id, task_id, ex_time, try_num)
+            else:
+                self.log.warning(
+                    "Unable to find Task in db directly for %s-%s (%s). Scan all tasks",
```

Comment:
can you emphasize that this can cause significant slowdowns?

Reply:
And add a link to k8s labeling guidelines?  https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Reply:
sure. I will rework the code a bit to check for this explicitly.